pallets/interest-accrual/src/lib.rs

```diff
@@ -226,11 +226,12 @@ pub mod pallet {
 			normalized_debt: T::Balance,
 		) -> Result<T::Balance, DispatchError> {
 			Rate::<T>::get(interest_rate_per_sec)
-				.ok_or(Error::<T>::NoSuchRate.into())
+				.ok_or(Error::<T>::NoSuchRate)
 				.and_then(|rate| {
 					Self::calculate_debt(normalized_debt, rate.accumulated_rate)
-						.ok_or(Error::<T>::DebtCalculationFailed.into())
+						.ok_or(Error::<T>::DebtCalculationFailed)
 				})
+				.map_err(Into::into)
```

Comment:
I think the other way to write this would be with the `?` operator:

```
let rate = Rate::<T>::get(...).ok_or(...)?;
let debt = Self::calculate_debt(...).ok_or(...)?;
Ok(debt)
```

I kept the combinators here since that was already the structure of this function, and just moved the `into` call to a single map_err` (which the `?` operator would do for us)